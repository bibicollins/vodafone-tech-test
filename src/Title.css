.title {
    left: 250px;
    top: 0px;
  width: 100%;

}
