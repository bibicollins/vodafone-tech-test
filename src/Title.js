import React, { Component } from 'react';
import PhoneData from './data/phones.json'
export class Title extends Component {
  render() {
    return (
      <div >
        {PhoneData.map((phone, index)=>{
          return <div className="title" key={index}>
          <h1>{phone.groupName}</h1>
          <h2>{phone.rating}</h2>
          </div>
        })}
        </div>
    );
  }
}

export default Title;
