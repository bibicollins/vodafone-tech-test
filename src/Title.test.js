import React from 'react';
import PhoneData from './data/phones.json'
import { mount, shallow } from 'enzyme';
import Title from './Title';

describe('Title', () => {
  let title = mount(<Title />);
  it('renders the title', () => {
    console.log(title.debug())
    expect(title.find('h1').text()).toEqual('Apple iPhone 8')
  });
  it('renders the rating', () => {
    expect(title.find('h2').text()).toEqual('4.2')
  });
  it('renders the first part of description', () => {
    expect(title.find('h3').text()).toEqual('All-glass design, advanced cameras, wireless charging and a smart A11 Bionic chip.')
  })
  it('renders the second part of description', () => {
    expect(title.find('h4').text()).toEqual('Intelligence never looked better.')
  })
});
