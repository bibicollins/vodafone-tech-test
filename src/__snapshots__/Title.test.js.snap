// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<div>
  <div
    className="title"
  >
    <h1>
      Apple iPhone 8
    </h1>
    <h2>
      4.2
    </h2>
    <h3>
      All-glass design, advanced cameras, wireless charging and a smart A11 Bionic chip.
    </h3>
    <h4>
      Intelligence never looked better.
    </h4>
  </div>
</div>
`;
